src/org/exist/dom/memtree/MemTreeBuilder.java

```diff
@@ -415,12 +416,26 @@ public int namespaceNode(final String prefix, final String uri) {
 
 
     public int namespaceNode(final QName qname) {
+        return namespaceNode(qname, false);
+    }
+
+    public int namespaceNode(final QName qname, boolean checkNS) {
         final int lastNode = doc.getLastNode();
         boolean addNode = true;
         if(doc.nodeName != null) {
             final QName elemQN = doc.nodeName[lastNode];
             if(elemQN != null) {
                 final String elemPrefix = (elemQN.getPrefix() == null) ? XMLConstants.DEFAULT_NS_PREFIX : elemQN.getPrefix();
+                if (checkNS
+                    && XMLConstants.DEFAULT_NS_PREFIX.equals(elemPrefix)
+                    && "xmlns".equals(qname.getPrefix())
+                    && "".equals(qname.getLocalPart())) {
```

Comment:
Sorry I made a copy-paste error. Should be:

```java
&&  XMLConstants.NULL_NS_URI.equals(qname.getNamespaceURI())
```
As checking the local part of a QName used for the namespace doesn't make any sense.